### ISSUE TITLE
Layout/EndAlignment should autocorrect

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#8321]
```
         def rspec_pattern
           @rspec_pattern ||= if rspec_pattern_config?
                               Regexp.union(rspec_pattern_config.map(&Regexp.public_method(:new)))
                             else
                               DEFAULT_PATTERN_RE
           end
         end
```

I get "Layout/EndAlignment: end at 52, 10 is not aligned with if at 48, 29."

The cop should autocorrect it.

**Comment by sebastian-palma** [comment]
If I try to autocorrect with the given example it does it but `AutoCorrect` in the `.rubocop.yml` must be set to true. What's not autocorrected is the content inside the if;

```bash
--- a/file.rb
+++ b/file.rb
@@ -49,5 +49,5 @@
                               Regexp.union(rspec_pattern_config.map(&Regexp.public_method(:new)))
                             else
                               DEFAULT_PATTERN_RE
-          end
+                             end
         end
```

**Comment by marcandre** [comment]
Right, the inside is another cop's responsibility.

The full story is that I added `@rspec_pattern ||= ` below:

```ruby
         def rspec_pattern
           @rspec_pattern ||= if rspec_pattern_config?
             Regexp.union(rspec_pattern_config.map(&Regexp.public_method(:new)))
           else
             DEFAULT_PATTERN_RE
           end
         end
```

RuboCop corrected the alignment of everything except the `end`. It's not the first time I notice this, but now I took the time to open this issue.

**Comment by sebastian-palma** [comment]
What's the previous and the autocorrected code @marcandre? I ran rubocop to autocorrect your example and got this:

```ruby
--- a/file.rb
+++ b/file.rb
@@ -49,5 +49,5 @@
             Regexp.union(rspec_pattern_config.map(&Regexp.public_method(:new)))
           else
             DEFAULT_PATTERN_RE
-          end
+                             end
         end
```

In this case only the `end` is aligned with the `if`. I don't know if I'm missing something.

**Comment by shekhar-patil** [comment]
@marcandre Yes, there is an issue in the alignment of the 'END'. I would like to pick this issue.

**Comment by marcandre** [comment]
> but `AutoCorrect` in the `.rubocop.yml` must be set to true

I'm sorry I skipped over this @sebastian-palma, that was the only issue. Now fixed.
